### ISSUE TITLE
[no-void]: allow void arrow functions

### ISSUE THREAD

**Issue by G-Rath** [eslint/eslint#13020]
**What rule do you want to change?**
`no-void`

**Does this change cause the rule to produce more or fewer warnings?**
Fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
Option - could also be implemented as standard behaviour depending on interpretation of the rules description.

**Please provide some example code that this change will affect:**

```js
const log = x => void console.log(x);
```

**What does the rule currently do for this code?**
Errors

**What will the rule do after it's changed?**
Not error

**Are you willing to submit a pull request to implement this change?**
Yes

-----

There are some [legitimate uses](https://gist.github.com/slikts/dee3702357765dda3d484d8888d3029e) for the void operator, one of which is even mentioned in the opening sentence of the `no-void` docs:

>  It can be used to ignore any side effects expression may produce

The description of the `no-void` rule to me seems to indicate that the rule is actually ideally about preventing the use of `void` to get a "pure" undefined back in the pre-ES5 days when undefined was mutable: 

> The common case of using void operator is to get a "pure" undefined value as prior to ES5 the undefined variable was mutable:

So it might be arguable that this is technically a bug? But I don't have a major preference on if it's an option or not :)

Based off #12688, which was closed for not following the template.

**Comment by yeonjuan** [comment]
@G-Rath 
Thanks for the proposal. 👍 
I think it makes sense. I labeled it `evaluating` to see what other team members think.

**Comment by ljharb** [comment]
`undefined` is still mutable, post-ES5, just not in the global scope.

**Comment by G-Rath** [comment]
While I know this is still marked as `evaluating`, I went ahead and implemented this [in a branch](https://github.com/G-Rath/eslint/tree/no-void-allow-arrow-functions) with the option name `allowAtStartOfConciseArrowFunctions` since it's such a small change, and I was working on some surrounding stuff anyway.

**Comment by G-Rath** [comment]
@yeonjuan @kaicataldo I feel like there was some interest given the thumbs up and previous issue.

I avoid poking as I figured with current world happenings + preparing for v7 the attention of the team was focused on higher priorities for the time being, but I do have an implementation ready to go that I can PR if that's ok?

Otherwise could disinterest be confirmed, or this issue reopened?

**Comment by kaicataldo** [comment]
@G-Rath I support this change, but it doesn't appear there enough support from the team as a whole this time around. Feel free to create a new issue with a proposal if you'd still like to advocate for this (the PR can serve as the proposal as well, if you don't mind filling out the relevant template.

**Comment by G-Rath** [comment]
@kaicataldo sure thing 🙂
